### ISSUE TITLE
how to ensure yolov3 is being trained correctly?

### ISSUE THREAD

**Issue by banuk2** [AlexeyAB/darknet#3251]
**./darknet detector train pklot.data cfg/yolov3_pklot.cfg yolov3.weights**

after initiating the training, only 3 lines of output is displayed as mentioned below. 
**Loading weights from yolov3.weights...Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Saving weights to backup/yolov3_pklot_final.weights**

I suspect model is not trained at all on the data provided, although weights are saved model is not detecting the objects or no bounding boxes are created.
 
Information related to training,
Image dataset has around 3500 images (no image augmentation) and also corresponding txt files placed across different sub-folders.

cfg -> classes =2 (in 3 [yolo]), filters = 21 (in 3 [convolutional]), batches = 64, sub division = 16
learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

I have tried same with yolov3-tiny but change in cfg according, still the same issue.

**Comment by AlexeyAB** [comment]
@banuk2 Hi,

You should use `darknet53.conv.74` instead of `yolov3.weights` https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

`./darknet detector train pklot.data cfg/yolov3_pklot.cfg darknet53.conv.74`


----

Also if you have Val/Test dataset that is specified `val=val.txt` in `pklot.data` file then it is better to train with flag `-map` so you will see accuracy (mAP) progress during training: https://github.com/AlexeyAB/darknet#when-should-i-stop-training

`./darknet detector train pklot.data cfg/yolov3_pklot.cfg darknet53.conv.74 -map`

**Comment by banuk2** [comment]
Thanks @AlexeyAB , darknet53.conv.74 weights have made the difference, training has been initiated now and logs are being generated. 

I would like to know why could not I train with yolov3.weights

**Comment by kevinrev26** [comment]
I use this to train YOLO with my own dataset: 
`/darknet detector train my_set.data cfg/yolov3_my_set.cfg darknet53.conv.74`

After reached the early stopping point. The detections are incorrect: My dataset is for guns, but it detectes people over another images.

What could be wrong?

(I also used the yolo cfg and the readme guide to train)

Thanks for advice.

**Comment by AlexeyAB** [comment]
Can you show wrong detections?
What command do you use for detection?
Show content of `my_set.data` and `obj.names` files.

**Comment by kevinrev26** [comment]
This is my wrong detection
![lol2](https://user-images.githubusercontent.com/12385068/58520692-306bb580-8176-11e9-81a5-7beac421ad64.jpg)

This is my guns.data
[guns.txt](https://github.com/AlexeyAB/darknet/files/3230228/guns.txt)

And my guns.names
[guns.names.txt](https://github.com/AlexeyAB/darknet/files/3230230/guns.names.txt)

Seems like i need to give time to the training?

**Comment by AlexeyAB** [comment]
@kevinrev26 
If you use the correct command for detection then it can't detect person regardless how well trained the model is.

It seems you use incorrect command for detection.
Check that you specified `my_set.data` in the detection command, and it refers to `names = /data/guns/guns.names` with correct path.
